tiledb/sm/array_schema/dimension.h

```diff
@@ -67,9 +68,6 @@ class Dimension {
   /*     CONSTRUCTORS & DESTRUCTORS    */
   /* ********************************* */
 
-  /** Constructor. */
-  Dimension();
-
```

Comment:
Can we just use Dimension()=delete; to explicitly disable the default constructor?

Reply:
Sure. That's a good idea.

Reply:
... and done.